conans/assets/templates/new_v2_autotools.py

```diff
@@ -48,6 +49,8 @@ def build(self):
         def package(self):
             autotools = Autotools(self)
             autotools.install()
+            if self.settings.get_safe("os") == "Macos" and self.options.get_safe("shared", False):
```

Comment:
I'd say that we want to put these conditionals inside the tool itself. This way, it keeps working for example in 2.0 when we use ``package_type`` to check for "shared", and any evolution like being enabled for iOS & family.